src/openloco/ui/dropdown.cpp

```diff
@@ -97,32 +115,338 @@ namespace openloco::ui::dropdown
+                            sub_494BF6(self, dpi, dropdownItemFormat, x, y, width, colour, args);
+                        }
+                    }
+
+                    if (dropdownItemFormat == (string_id)-2 || dropdownItemFormat == string_ids::null)
+                    {
+                        auto x = _windowDropdownOnpaintCellX * _dropdownItemWidth + self->x + 2;
+                        auto y = _windowDropdownOnpaintCellY * _dropdownItemHeight + self->y + 2;
+
+                        auto imageId = *(uint32_t*)&args;
+                        if (dropdownItemFormat == (string_id)-2 && itemCount == _dropdownHighlightedIndex)
+                        {
+                            imageId++;
+                        }
+                        gfx::draw_image(dpi, x, y, imageId);
+                    }
+                }
+                else
+                {
+                    auto x = _windowDropdownOnpaintCellX * _dropdownItemWidth + self->x + 2;
+                    auto y = _windowDropdownOnpaintCellY * 3 * (_dropdownItemHeight / 2) + self->y + 1;
+
+                    if (!(self->colours[0] & colour::translucent_flag))
+                    {
+                        gfx::draw_rect(dpi, x, y, _dropdownItemWidth - 1, 0, colour::get_shade(self->colours[0], 3));
+                        gfx::draw_rect(dpi, x, y + 1, _dropdownItemWidth - 1, 0, colour::get_shade(self->colours[0], 7));
+                    }
+                    else
+                    {
+                        auto colour = _byte_50457A[colour::opaque(self->colours[0])] | (1 << 25);
+                        colour++;
+                        gfx::draw_rect(dpi, x, y, _dropdownItemWidth - 1, 0, colour);
+                        colour++;
+                        gfx::draw_rect(dpi, x, y + 1, _dropdownItemWidth - 1, 0, colour);
+                    }
+                }
+
+                _windowDropdownOnpaintCellX++;
+                if (_windowDropdownOnpaintCellX >= _dropdownColumnCount)
+                {
+                    _windowDropdownOnpaintCellX = 0;
+                    _windowDropdownOnpaintCellY++;
+                }
+            }
+        }
+
+        static void initEvents()
+        {
+            events.on_update = onUpdate;
+            events.draw = draw;
+        }
+
+        // 0x004CCF1E
+        static void open(gfx::point_t origin, gfx::ui_size_t size, colour_t colour)
+        {
+            auto window = WindowManager::createWindow(WindowType::dropdown, origin, size, window_flags::stick_to_front, &common::events);
+
+            window->widgets = common::widgets;
+
+            if (colour & colour::translucent_flag)
+            {
+                window->flags |= window_flags::transparent;
+            }
+
+            common::initEvents();
+
+            common::widgets[0].colour = colour::black;
+            window->colours[0] = colour;
+
+            _dropdownHighlightedIndex = -1;
```

Comment:
Removed the reseting of the dropdownItemFormats to zero. Assume that is because not all callers did this.

Reply:
Yep `showText2` doesn't reset the dropdownItemFormats